File: tide-canvas-web/src/types/redeem.ts
import type { PageQuery } from "./api";

export interface RedeemCodeVO {
  id: number;
  code: string;
  points: number;
  /** 生成者（管理员）用户ID */
  createdBy?: number;
  /** 0未使用 / 1已使用 / 2已停用 */
  status: number;
  usedBy?: number;
  usedTime?: string;
  expireTime?: string;
  batchNo?: string;
  remark?: string;
  createTime: string;
}

export interface RedeemCodeQuery extends PageQuery {
  code?: string;
  status?: number;
  batchNo?: string;
}

export interface RedeemResultVO {
  points: number;
  balance: number;
}

export interface GenerateRedeemDTO {
  count: number;
  points: number;
  expireTime?: string;
  remark?: string;
}

File: tide-canvas-go/internal/model/redeem.go
package model

import "time"

// RedeemCode 兑换码表 redeem_code（code 即对外凭证，无 public_id）。
type RedeemCode struct {
	SoftDeleteModel
	Code       string     `json:"code" gorm:"column:code"`
	Points     int        `json:"points" gorm:"column:points"`
	CreatedBy  *int64     `json:"-" gorm:"column:created_by"`
	Status     int        `json:"status" gorm:"column:status"`
	UsedBy     *int64     `json:"-" gorm:"column:used_by"`
	UsedTime   *time.Time `json:"usedTime" gorm:"column:used_time"`
	ExpireTime *time.Time `json:"expireTime" gorm:"column:expire_time"`
	BatchNo    string     `json:"batchNo" gorm:"column:batch_no"`
	Remark     string     `json:"remark" gorm:"column:remark"`
}

// TableName 表名。
func (RedeemCode) TableName() string { return "redeem_code" }
